classes/CartRule.php

```diff
@@ -1206,8 +1206,8 @@ public function getContextualValue($use_tax, Context $context = null, $filter =
                 $selected_products = $this->checkProductRestrictionsFromCart($context->cart, true);
                 if (is_array($selected_products)) {
                     foreach ($package_products as $product) {
-                        if (in_array($product['id_product'] . '-' . $product['id_product_attribute'], $selected_products)
-                            || in_array($product['id_product'] . '-0', $selected_products)
+                        if ((in_array($product['id_product'] . '-' . $product['id_product_attribute'], $selected_products)
```

Comment:
Because it's not really clear, can you clear this condition :sweat_smile: 
can you
```suggestion
if (
    (
        in_array($product['id_product'] . '-' . $product['id_product_attribute'], $selected_products)
        || in_array($product['id_product'] . '-0', $selected_products)
    )
    && (
        (
            $this->reduction_exclude_special
            && !$product['reduction_applies']
        )
        || !$this->reduction_exclude_special
    )
) {
```

Reply:
if the first in_array function returns true, the rest of the condition after the OR is ignored

Reply:
I know, I just suggest for a clean 😅

Reply:
sorry i am a beginner on github 🤭

Reply:
we all started one day 😉thank you for trying ! ❤️